### PR TITLE
[Fix] fix RandomGrayscale bug, input image should be bgr format.

### PR DESCRIPTION
**Pull request by Fei-Wang**
## Motivation

Fix RandomGrayscale bug.

## Modification

change rgb2gray to bgr2gray.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by Fei-Wang** [comment]
@mzr1996

**Comment by Ezra-Yu** [comment]
Can you describe under what circumstances the bug will occur?

File: mmcls/datasets/pipelines/transforms.py [hunk 1]
```diff
@@ -419,7 +419,7 @@ def __call__(self, results):
             num_output_channels = img.shape[2]
             if random.random() < self.gray_prob:
                 if num_output_channels > 1:
-                    img = mmcv.rgb2gray(img)[:, :, None]
+                    img = mmcv.bgr2gray(img, keepdim=True)
```
Comment (mzr1996) [inline mmcls/datasets/pipelines/transforms.py:422]: I think we cannot change the behavior directly. The `cvtColor` of OpenCV has different weights of different channels. This modification may introduce BC-breaking changes.
A better method is to add an argument, maybe `color_format`, to change the behavior, and the default value should be `rgb` to keep the original behavior.

**Comment by Fei-Wang** [comment]
Because in general data aug is earlier than norm, and data aug is in bgr format.
If you follow the current practice, you need to bgr2rgb before grayscale, and then rgb2bgr after it is done (although grayscale does not need it), because norm generally does bgr2rgb by default.
https://github.com/open-mmlab/mmclassification/blob/c94e9b36699094ca8cfd975442474ca4ac47f083/configs/_base_/datasets/imagenet_bs64_pil_resize_autoaug.py#L17-L18

**Comment by mzr1996** [comment]
> 

Yes, I know it's before normalization. But for some datasets, like CIFAR, the images are not loaded from files, and these images are in RGB format.

**Comment by Fei-Wang** [comment]
> Yes, I know it's before normalization. But for some datasets, like CIFAR, the images are not loaded from files, and these images are in RGB format.

Thanks for your reply, you are right. I just have a question, because the current situation is that the channel order is different when apply aug, Why not  image processed same when it is loaded?

Now you are free to close this pr.

**Pull request closed without merging**